Extensions/dnSpy.AsmEditor/UndoRedo/Commands.cs

```diff
@@ -47,11 +52,15 @@ sealed class UndoRedoCommmandLoader : IAutoLoaded {
 			var count = undoCommandService.Value.NumberOfModifiedDocuments;
 			if (count == 0)
 				return;
-
 			var msg = count == 1 ? dnSpy_AsmEditor_Resources.AskExitUnsavedFile :
 					string.Format(dnSpy_AsmEditor_Resources.AskExitUnsavedFiles, count);
 			var res = messageBoxService.Show(msg, MsgBoxButton.Yes | MsgBoxButton.No);
-			if (res == MsgBoxButton.No || res == MsgBoxButton.None)
+            if (res == MsgBoxButton.Yes) {
+                appWindow.MainWindow.Hide();
```

Comment:
I think it is more intuitive to have a main window closed after a closing action and to let a user finish file saving in a self-contained window. You cannot return from there to the main window anyway, it is about to close, and closing the main window as a result of accepting, canceling, or exiting the save window seems to be a little bit "transected". What do you think?